packages/rsbuild-plugin-angular/src/lib/plugin/utils/component-resolvers.ts

```diff
@@ -58,7 +54,10 @@ export class StyleUrlsResolver {
     // resulting in the resolver being called multiple times. While the code changes, the
     // `styleUrls` may remain constant, which means we should always return the previously
     // resolved style URLs.
-    if (entry && entry.matchedStyleUrls === matchedStyleUrls) {
+    if (
+      entry &&
+      entry.matchedStyleUrls.join(',') === matchedStyleUrls.join(',')
```

Comment:
This condition was never met before. I am not 100% sure if this was a bug or there is a case where the condition `entry && entry.matchedStyleUrls === matchedStyleUrls` would be true.

Reply:
My assumption here is that it is related to rebuilds - where there is a cache of existing transformed urls